Diff format should not include pointer addresses

The diff format used by `assert.Equal()` etc. results in a lot of false positives because values are printed with pointer addresses. Here's an example:

```
=== RUN   TestBuilder_BuildRuleForAttribute_extendedMapType_withAttrOverrides
--- FAIL: TestBuilder_BuildRuleForAttribute_extendedMapType_withAttrOverrides (0.00s)
        Error Trace:    builder_test.go:972
    Error:      Not equal: &validation.AttributeCheck{Name:"homeAddress", Rule:(*validation.And)(0xc82016eca0)} (expected)
                    != &validation.AttributeCheck{Name:"homeAddress", Rule:(*validation.And)(0xc82016ec20)} (actual)

            Diff:
            --- Expected
            +++ Actual
            @@ -1,15 +1,8 @@
            -(*validation.AttributeCheck)(0xc82016ecc0)({
            +(*validation.AttributeCheck)(0xc82016ec40)({
              Name: (string) (len=11) "homeAddress",
            - Rule: (*validation.And)(0xc82016eca0)({
            + Rule: (*validation.And)(0xc82016ec20)({
               Rules: ([]validation.Rule) (len=4 cap=4) {
            -   (*validation.AttributeCheck)(0xc82016ec60)({
            +   (*validation.AttributeCheck)(0xc82016eb80)({
                 Name: (string) (len=5) "state",
            -    Rule: (*validation.BuiltinTypecheck)(0xc82016dcd0)({
            -     Type: (types.integerType) {
            -     }
            -    })
            -   }),
            -   (*validation.AttributeCheck)(0xc82016ec80)({
            -    Name: (string) (len=13) "streetAddress",
            -    Rule: (*validation.BuiltinTypecheck)(0xc82016dce0)({
            +    Rule: (*validation.BuiltinTypecheck)(0xc82016dcb0)({
                  Type: (types.stringType) {
            @@ -18,3 +11,10 @@
                }),
            -   (*validation.UnknownKeys)(0xc82008a2b8)({
            +   (*validation.AttributeCheck)(0xc82016eba0)({
            +    Name: (string) (len=13) "streetAddress",
            +    Rule: (*validation.BuiltinTypecheck)(0xc82016dcc0)({
            +     Type: (types.stringType) {
            +     }
            +    })
            +   }),
            +   (*validation.UnknownKeys)(0xc82008a2b0)({
                 keys: (map[string]struct {}) (len=3) {
            @@ -28,3 +28,3 @@
                }),
            -   (*validation.BuiltinTypecheck)(0xc82016dcf0)({
            +   (*validation.BuiltinTypecheck)(0xc82016dba0)({
                 Type: (types.mapType) {        
```

I'm referring to this:

```
            -(*validation.AttributeCheck)(0xc82016ecc0)({
            +(*validation.AttributeCheck)(0xc82016ec40)({
```

Considering that `Equal` always compares by _value_, knowing the pointer addresses is generally not useful or desirable.

Spew has a format `&#v` that we could use instead of `spew.Sdump()`.

--- a/Godeps/Godeps.json
+++ b/Godeps/Godeps.json
@@ -9,7 +9,7 @@
 		{
 			"ImportPath": "github.com/davecgh/go-spew/spew",
 			"Comment": "v1.0.0-3-g6d21280",
-			"Rev": "6d212800a42e8ab5c146b8ace3490ee17e5225f9"
+			"Rev": "04cdfd42973bb9c8589fd6a731800cf222fde1a9"
 		},
 		{
 			"ImportPath": "github.com/pmezard/go-difflib/difflib",

--- a/assert/assertions.go
+++ b/assert/assertions.go
@@ -18,10 +18,6 @@ import (
 	"github.com/pmezard/go-difflib/difflib"
 )
 
-func init() {
-	spew.Config.SortKeys = true
-}
-
 // TestingT is an interface wrapper around *testing.T
 type TestingT interface {
 	Errorf(format string, args ...interface{})
@@ -1043,8 +1039,8 @@ func diff(expected interface{}, actual interface{}) string {
 		return ""
 	}
 
-	e := spew.Sdump(expected)
-	a := spew.Sdump(actual)
+	e := spewConfig.Sdump(expected)
+	a := spewConfig.Sdump(actual)
 
 	diff, _ := difflib.GetUnifiedDiffString(difflib.UnifiedDiff{
 		A:        difflib.SplitLines(e),
@@ -1057,4 +1053,11 @@ func diff(expected interface{}, actual interface{}) string {
 	})
 
 	return "\n\nDiff:\n" + diff
+}
+
+var spewConfig = spew.ConfigState{
+	Indent:                  " ",
+	DisablePointerAddresses: true,
+	DisableCapacities:       true,
+	SortKeys:                true,
 }

--- a/mock/mock.go
+++ b/mock/mock.go
@@ -15,10 +15,6 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func inin() {
-	spew.Config.SortKeys = true
-}
-
 // TestingT is an interface wrapper around *testing.T
 type TestingT interface {
 	Logf(format string, args ...interface{})
@@ -746,8 +742,8 @@ func diff(expected interface{}, actual interface{}) string {
 		return ""
 	}
 
-	e := spew.Sdump(expected)
-	a := spew.Sdump(actual)
+	e := spewConfig.Sdump(expected)
+	a := spewConfig.Sdump(actual)
 
 	diff, _ := difflib.GetUnifiedDiffString(difflib.UnifiedDiff{
 		A:        difflib.SplitLines(e),
@@ -760,4 +756,11 @@ func diff(expected interface{}, actual interface{}) string {
 	})
 
 	return diff
+}
+
+var spewConfig = spew.ConfigState{
+	Indent:                  " ",
+	DisablePointerAddresses: true,
+	DisableCapacities:       true,
+	SortKeys:                true,
 }

